aFWall/src/main/java/dev/ukanth/ufirewall/util/LogNetUtil.java

```diff
@@ -0,0 +1,307 @@
+
+            }catch (Exception e){
+
+                Log.e(TAG, "Exception(03): " + e.getMessage());
+            }
+
+            return context.getString(R.string.error_or_unknown_category);
+        }
+
+        @Override
+        protected void onPostExecute(String s) {
+            super.onPostExecute(s);
+
+            if(onFinishRequest_callback != null){
+                onFinishRequest_callback.then(s);
+            }
+
+            try {
+                if ((progress != null) && progress.isShowing()) {
+                    progress.dismiss();
+                }
+            } catch (IllegalArgumentException e) {
+                Log.e(TAG, e.getMessage());
+                // Handle or log or ignore
+            } catch (final Exception e) {
+                Log.e(TAG, e.getMessage());
+                // Handle or log or ignore
+            } finally {
+                progress = null;
+            }
+
+            output_result = s;
+
+            new MaterialDialog.Builder(context)
+                    .content(s)
+                    .title(R.string.result)
+                    .neutralText(R.string.OK)
+                    .positiveText(R.string.copy_text)
+                    .onPositive(new MaterialDialog.SingleButtonCallback() {
+                        @Override
+                        public void onClick(@NonNull MaterialDialog dialog, @NonNull DialogAction which) {
+                            copyToClipboard(context, output_result);
+                            Api.toast(context, context.getString(R.string.result_copied_to_clipboard));
+                        }
+                    })
+                    .show();
+        }
+
+        String normal_ping(String ip){
+
+            String result = "";
+
+            try {
+
+                if(InetAddress.getByAddress(ip.getBytes()).isReachable(G.logPingTimeout() * 1000)){ // isReachable expect timeout in millisecond
+
+                    result = String.format(context.getString(R.string.reachable_timeout), finish_time());
+                }
+
+            } catch (IOException e) {
+
+                Log.e(TAG, "Exception(04): " + e.getMessage());
+
+                result = String.format("Currently IP(%s) is not Reachable, timeout: %d ms", ip, finish_time());
+            }
+
+            return result;
+        }
+
+        String su_busyboox_ping(String ip){
+
+            // using libsuperuser to perform ping by Busybox,
+            // This will need permission in AFWall+
+            // "0:(root) Apps running as root"
+
+            String result = "";
+
+            String command = String.format(PING_CMD, Api.getBusyBoxPath(context, true), G.logPingTimeout(), ip);
+
+            Log.d(TAG, "Execute CMD: " + command);
+
+            result = parse(Shell.run("su", new String[]{
+                    command
+            }, null, true));
+
+            if(result.isEmpty()){
+
+                return context.getString(R.string.network_connection_not_available);
```

Comment:
@vzool , If result is empty, you are saying network not available and other place you are saying access denied ?